nifi-toolkit/nifi-toolkit-tls/src/main/java/org/apache/nifi/toolkit/tls/manager/TlsClientManager.java

```diff
@@ -0,0 +1,116 @@
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableEntryException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class TlsClientManager extends BaseTlsManager {
+    private final TlsClientConfig tlsClientConfig;
+    private final KeyStore trustStore;
+    private final List<ConfigurationWriter<TlsClientConfig>> configurationWriters;
+    private final Set<String> certificateAliases;
+    private File certificateAuthorityDirectory;
+
+    public TlsClientManager(TlsClientConfig tlsClientConfig) throws GeneralSecurityException, IOException {
+        this(tlsClientConfig, new PasswordUtil(), FileInputStream::new);
+    }
+
+    public TlsClientManager(TlsClientConfig tlsClientConfig, PasswordUtil passwordUtil, InputStreamFactory inputStreamFactory) throws GeneralSecurityException, IOException {
+        super(tlsClientConfig, passwordUtil, inputStreamFactory);
+        this.trustStore = loadKeystore(tlsClientConfig.getTrustStore(), tlsClientConfig.getTrustStoreType(), tlsClientConfig.getTrustStorePassword());
+        this.tlsClientConfig = tlsClientConfig;
+        this.configurationWriters = new ArrayList<>();
+        this.certificateAliases = new HashSet<>();
+    }
+
+    public void setCertificateEntry(String alias, Certificate cert) throws KeyStoreException {
+        trustStore.setCertificateEntry(alias, cert);
+        certificateAliases.add(alias);
+    }
+
+    public void setCertificateAuthorityDirectory(File certificateAuthorityDirectory) {
+        this.certificateAuthorityDirectory = certificateAuthorityDirectory;
+    }
+
+    @Override
+    public void write(OutputStreamFactory outputStreamFactory) throws IOException, CertificateException, NoSuchAlgorithmException, KeyStoreException {
+        super.write(outputStreamFactory);
+
+        String trustStorePassword = tlsClientConfig.getTrustStorePassword();
+        if (StringUtils.isEmpty(trustStorePassword)) {
+            trustStorePassword = getPasswordUtil().generatePassword();
+            tlsClientConfig.setTrustStorePassword(trustStorePassword);
+        }
+
+        try (OutputStream outputStream = outputStreamFactory.create(new File(tlsClientConfig.getTrustStore()))) {
+            trustStore.store(outputStream, trustStorePassword.toCharArray());
+        }
+
+        for (ConfigurationWriter<TlsClientConfig> configurationWriter : configurationWriters) {
+            configurationWriter.write(tlsClientConfig);
+        }
+
+        if (certificateAuthorityDirectory != null) {
```

Comment:
Intended to address a possible future case where there are multiple trusted CAs to import into a browser